codalab/worker/worker_run_state.py

```diff
@@ -406,7 +408,17 @@ def check_and_report_finished(run_state):
                 finished=finished, exitcode=exitcode, failure_message=failure_msg
             )
 
-        def check_resource_utilization(run_state):
+        def check_resource_utilization(run_state: RunState):
+            try:
+                cpu_usage, memory_usage = docker_utils.get_container_stats_on_mac(
+                    run_state.container
+                )
+                run_state = run_state._replace(cpu_usage=cpu_usage)
+                run_state = run_state._replace(memory_usage=memory_usage)
+            except KeyError:
+                # If the bundle has exited, then some of the keys are missing, so we just dismiss it.
```

Comment:
Where does this `KeyError` come from?

Reply:
It comes from `docker_utils.get_container_stats_on_mac`. If the bundle has exited, for example, the `container_stats['memory_stats']` is an empty dict, and some other keys are also missing.

Reply:
Can't you handle this exception inside `get_container_stats_on_mac` then?